ground/uavobjgenerator/generators/flight/uavobjectgeneratorflight.cpp

```diff
@@ -272,9 +272,15 @@ bool UAVObjectGeneratorFlight::process_object(ObjectInfo* info)
             {
                 if ( info->fields[n]->type == FIELDTYPE_ENUM )
                 {
+                    int defaultVal;
+                    if (info->fields[n]->parent != NULL)
+                        defaultVal = info->fields[n]->parent->options.indexOf( info->fields[n]->defaultValues[0] );
+                    else
+                        defaultVal = info->fields[n]->options.indexOf( info->fields[n]->defaultValues[0] );
+
```

Comment:
This code snippet is repeated 4 times. It might make sense to look at the generators and see how much code is being duplicated and if we can move functions like this into a parent class.